src/firmware/guest/types/snp.rs

```diff
@@ -117,14 +113,76 @@ bitfield! {
 /// migration agent associated with it, the REPORT_ID_MA is filled in with the report ID of the
 /// migration agent.
 ///
-/// The firmware signs the attestation report with its VCEK. The firmware uses the system wide
+/// The firmware signs the attestation report with its VEK (VCEK or VLEK). The firmware uses the system wide
 /// ReportedTcb value as the TCB version to derive the VCEK. This value is set by the hypervisor.
+/// The VLEK is generated externally and has to be loaded into the machine.
 ///
 /// The firmware guarantees that the ReportedTcb value is never greater than the installed TCB
 /// version
+///
+/// Since the release of the 1.56 ABI, the Attestation Report was bumped from version 2 to 3.
+/// Due to content differences, both versions are kept separately in order to provide backwards compatibility and most reliable security.
+pub enum AttestationReport {
```

Comment:
```suggestion
pub enum AttestationReport {
```

I think this is probably the best way to do things given that we can't reasonably expect a user to know which version they're getting.